File: backend/InterviewManagement.API/Models/Question.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace InterviewManagement.API.Models
{
    public class Question
    {
        public Guid Id { get; set; }

        public string Title { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public string Answer { get; set; } = string.Empty;

        public Guid CreatedBy { get; set; }

        [ForeignKey(nameof(CreatedBy))]
        public User CreatedByUser { get; set; } = null!;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public bool IsDeleted { get; set; } = false;

        public ICollection<QuestionMetadata> QuestionMetadata { get; set; } = new List<QuestionMetadata>();
    }
}

File: frontend/interview-management-ui/src/types/question.ts
export interface Question {
  id: string;

  title: string;

  description: string;

  answer: string;

  topic: string;

  difficulty: string;

  isActive: boolean;

  createdBy: string;

  createdByName: string;

  createdAt: string;

  updatedAt?: string;
}